### PR TITLE
Cleanup of some manual notify calls

### PR DESCRIPTION
**Pull request by distantcam**
While reviewing adding more reactive stuff I noticed a few bad `NotifyOfPropertyChange` calls.

### CODE REVIEW

File: src/ServiceInsight/MessageList/MessageListViewModel.cs [hunk 1]
```diff
@@ -76,7 +75,9 @@ public class MessageListViewModel : RxConductor<RxScreen>.Collection.AllActive,
 
         public MessageSelectionContext Selection { get; }
 
-        public bool WorkInProgress => workCount > 0 && !Parent.AutoRefresh;
+        public int WorkCount { get; private set; }
```
Comment (HEskandari) [inline src/ServiceInsight/MessageList/MessageListViewModel.cs]: The previous implementation had this `WorkCount` as internal. While it is made public is it used anywhere? AFAIK all we bind to is `WorklInProgress` propery.

File: src/ServiceInsight/MessageList/MessageListViewModel.cs [hunk 2]
```diff
@@ -36,7 +36,6 @@ public class MessageListViewModel : RxConductor<RxScreen>.Collection.AllActive,
         GeneralHeaderViewModel generalHeaderDisplay;
         string lastSortColumn;
         bool lastSortOrderAscending;
-        int workCount;
```
Comment (adamralph) [inline src/ServiceInsight/MessageList/MessageListViewModel.cs]: This change seems redundant, and unrelated to introducing `nameof`

File: src/ServiceInsight/MessageList/MessageListViewModel.cs (hunk 1 above)
Comment (distantcam) [inline src/ServiceInsight/MessageList/MessageListViewModel.cs]: Yep. I thought I had to make it public for the Fody magic to work, but it turns out it works on private properties too. I'll modify the PR.

File: src/ServiceInsight/MessageList/MessageListViewModel.cs (hunk 2 above)
Comment (distantcam) [inline src/ServiceInsight/MessageList/MessageListViewModel.cs]: This PR isn't just about introducing nameof. It's also about removing unnecessary `NotifyOfPropertyChange` calls. I plan on eliminating as many as I can, but that's enough for this PR.

As for this change, it's not redundant. By making it a property `PropertyChanged.Fody` weaves in the change notification automatically. `PropertyChanged.Fody` also detects any properties using `WorkCount` like `WorkInProgress` and automatically fires a notification for that property too. That's how I can eliminate the manual `NotifyOfPropertyChange(WorkInProgress)` calls.

Comment (adamralph) [inline src/ServiceInsight/MessageList/MessageListViewModel.cs]: Ah, IL weaving magic 😄 

Thanks for the explanation.

**Comment by adamralph** [comment]
@HEskandari do you have any remaining concerns?

**Comment by HEskandari** [comment]
@adamralph yes :) I initially thought the private field is better, but had a chat with @distantcam and we agreed that a public property is a better choice here. The reasoning is that for someone new to the code it is hard to understand why a private property is needed. Also the tooling (e.g. Resharper and alikes) would wrongly suggest you to remove it.

**Comment by distantcam** [comment]
@HEskandari You meant private property, not private field. @adamralph The discussion Hadi and I had was on private vs public, not on field vs property.

**Comment by HEskandari** [comment]
@distantcam yes, correct. Sorry for the confusion.

We decided to change it back to public property AFAIR and not a field.

**Comment by distantcam** [comment]
@HEskandari Yep. I've rolled back the last commit, and rebrached everything onto latest master.

**Comment by HEskandari** [comment]
@distantcam cool, I'm seeing it now.

**Comment by adamralph** [comment]
Discussed with @HEskandari. He will smoke test this and if all OK, merge it.

**Comment by HEskandari** [comment]
@adamralph I just got to this today. I've smoke tested and things are working as expected.

**Pull request merged**